### PR TITLE
Set proper cache dir for posix build

### PR DESCRIPTION
**Pull request by toots**
This PR enhances caching support by separating `system` vs. `user` cache. 

System cache is mostly for stdlib caching and should be shared by all users while user cache is for user scripts and should be user-specific.

Default locations:
- windows: `<executable directory>/cache` for both
- posix (system-wide install):
  - `system`: `/var/cache/liquidsoap`
  - `user`: `$HOME/.cache/liquidsoap`
- dune:
  - `system`: set by dune
    - Local build: `_build/install/default/share/liquidsoap-lang/cache` 
    - Installed: `~/.opam/<switch>/share/liquidsoap-lang/cache`
  - `user`: `$HOME/.cache/liquidsoap`

Fixes: #3970

### CODE REVIEW

**Comment by toots** [comment]
@vitoyucepi Description and PR updated. This should cover all bases now.

**Comment by vitoyucepi** [comment]
I didn't ask this question before, will the cache contain the hardcoded passwords?
```ruby
password = "pass"
output.icecast(%mp3, blank(), password=password, mount="1")
output.icecast(%mp3, blank(), password="pass", mount="2")
```

**Comment by toots** [comment]
> I didn't ask this question before, will the cache contain the hardcoded passwords?
> 
> ```ruby
> password = "pass"
> output.icecast(%mp3, blank(), password=password, mount="1")
> output.icecast(%mp3, blank(), password="pass", mount="2")
> ```

Excellent catch, thank you.

I just added proper permissions with customization env variables and documented them and also security concerns in the doc.

File: doc/content/language.md [hunk 1]
```diff
@@ -1434,34 +1434,82 @@ During the first execution, the script is parsed, type checked and evaluated. On
...
+The following environment variables control the cache behavior:
 
-Thus, if you need to change the defaults and run the cache maintenance, you can configure the values in your script and run `liquidsoap.cache.maintenance()` manually.
+- `LIQ_CACHE`: disable the cache when set to anything else than `1` or `true`
+- `LIQ_CACHE_SYSTEM_DIR`: set the cache system directory
+- `LIQ_CACHE_SYSTEM_DIR_PERMS`: set the permission used when creating cache system directory (and its parents when needed). Default: `0o755`
+- `LIQ_CACHE_SYSTEM_FILE_PERMS`: set the permissions used when creating a system cache file. Default: `0o644`
+- `LIQ_CACHE_USER_DIR`: set the cache user directory
+- `LIQ_CACHE_USER_DIR_PERMS`: set the permission used when creating cache user directory (and its parents when needed). Default: `0o700`.
+- `LIQ_CACHE_USER_FILE_PERMS`: set the permissions used when creating a user cache file. Default: `0o644`
```
Comment (vitoyucepi) [inline doc/content/language.md:1513]:
```suggestion
- `LIQ_CACHE_USER_FILE_PERMS`: set the permissions used when creating a user cache file. Default: `0o600`
```

**Comment by toots** [comment]
I think that this is good for merge. The docker build issues are related to debian-multimedia changing their DNS I believe.

**Pull request merged**